packages/lib/hooks/useKeyPress.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect, RefObject, useRef } from "react";
+
+export function useKeyPress(targetKey: string, ref?: RefObject<HTMLInputElement>): boolean {
+  // State for keeping track of whether key is pressed
+  const [keyPressed, setKeyPressed] = useState(false);
+  const placeHolderRef = ref?.current;
+  // If pressed key is our target key then set to true
+  function downHandler({ key }: { key: string }): void {
+    if (key === targetKey) {
+      setKeyPressed(true);
+    }
+  }
+  // If released key is our target key then set to false
+  const upHandler = ({ key }: { key: string }): void => {
+    if (key === targetKey) {
+      setKeyPressed(false);
+    }
+  };
+  // Add event listeners
+  useEffect(() => {
+    if (ref && placeHolderRef) {
+      placeHolderRef.addEventListener("keydown", downHandler);
+      placeHolderRef.addEventListener("keyup", upHandler);
+      return () => {
+        placeHolderRef?.removeEventListener("keydown", downHandler);
+        placeHolderRef?.removeEventListener("keyup", upHandler);
+      };
+    } else {
+      window.addEventListener("keydown", downHandler);
+      window.addEventListener("keyup", upHandler);
+      // Remove event listeners on cleanup
+      return () => {
+        window.removeEventListener("keydown", downHandler);
+        window.removeEventListener("keyup", upHandler);
+      };
```

Comment:
Handle keypress on the red or when in a window 

Usefull if you only wanna do stuff when something is focused